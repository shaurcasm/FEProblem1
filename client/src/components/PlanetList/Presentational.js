import React, { useEffect, useState } from 'react';
import './style.scss';

// Any datalist styling; 
const Presentational = ({ planets, select, changeImage }) => {
    const [selectedPlanet, setSelection] = useState('');
    const [previousSelection, setPrevious] = useState('');

    // Valuable code. Validating Datalist input.
    // When the value of the input changes...
    const inputValidation = event => {
        var optionFound = false,
            input = event.target,
            datalist = input.list;
        
        // Determine whether an option exists with the current value of the input.
        for (let j = 0; j < datalist.options.length; j++) {
            let optionValue = datalist.options[j].value;
            if(input.value.toUpperCase() === optionValue.toUpperCase()) {
                optionFound = true;
                setSelection(optionValue);
                changeImage(optionValue);
                break;
            }
        }
        // Use the setCustomValidity function of the validation API
        // to provide a user feedback if the value does not exist in the datalist.
        if(optionFound) {
            input.setCustomValidity('');
        } else {
            input.setCustomValidity('Please select a valid Planet');
        }
        input.reportValidity();
    }

    // Apply if local state selected planet is changed(second argument)
    useEffect(() => {
        //console.log("Current Planet: ", selectedPlanet);
        //console.log("Previous Planet: ", previousSelection);

        if(selectedPlanet !== previousSelection) {
            // If there was a previous selection, replace it with new one use redux action
            if(previousSelection) {
                //console.log('Replace action!');
                select.replacePlanet(selectedPlanet, previousSelection);
                setPrevious(selectedPlanet);
            }
            // Else, no previous selection. Merely use add action.
            else {
                //console.log('Add action!')
                select.addPlanet(selectedPlanet);
                setPrevious(selectedPlanet);
            }
        }
    }, [previousSelection, select, selectedPlanet]);

    // Gets updated everytime planets state is changed.
    const planetOptions = planets.filter(planet => planet.selected === false).map((planet, index) => {
        return (
            <option key={index} value={planet.name}>{planet.name}</option>
        )        
    });
    
    // Comment: PlanetList component's datalist didn't need dynamic ID to function properly, for currently unknown reason.
    // Unlike VehicleList, yet they both were applied once by multiple sectional components.
    return (
        <div className='planet-list'>
            <label htmlFor='planet-selector' className='sr-only'>Choose a Planet from this list:</label>
            <input type='text' list='available-planets' name='planet-selector' onChange={inputValidation} placeholder="Pick a Planet." autoCorrect='off' spellCheck='false' autoFocus />
            <datalist id='available-planets'>
                {planetOptions}
            </datalist>
        </div>
    );
}

export default Presentational;