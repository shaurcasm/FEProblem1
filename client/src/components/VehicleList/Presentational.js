import React, { useEffect, useState } from 'react';
import './style.scss';

const Presentational = ({ vehicles, select, distance }) => {
    const [selectedVehicle, setSelection] = useState('');
    const [previousSelection, setPrevious] = useState('');
    const [options, setOptions] = useState([]); // To Keep track of options even after re-renders.

    // Update/Re-render options everytime the distance Props or Redux state Vehicles are changed.
    useEffect(() => {
      //console.log("Distance to Planet: ", distance);
      setOptions([])
      var tempOptions = [];
      vehicles.forEach(vehicle => {
        if((vehicle['total_no'] > 0) && (vehicle['max_distance'] >= distance)) {
          let option = 
          <option key={vehicle["max_distance"]} value={vehicle.name}>
            {vehicle.name} X {vehicle["total_no"]}
          </option>;
          tempOptions.push(option);
        }  
      });
      setOptions([...tempOptions]);
    }, [distance, vehicles]);

    // Apply if local state selected planet is changed(second argument)
    useEffect(() => {
        //console.log("Current Vehicle: ", selectedVehicle);
        //console.log("Previous Vehicle: ", previousSelection);
        //console.log("Vehicle array: ", localVehicles);

        if(selectedVehicle !== previousSelection) {
            // If there was a previous selection, replace it with new one use redux action
            if(previousSelection) {
                console.log('Replace action!');
                select.replaceVehicle(selectedVehicle, previousSelection);
                setPrevious(selectedVehicle);
            }
            // Else, no previous selection. Merely use add action.
            else {
                console.log('Add action!')
                select.addVehicle(selectedVehicle);
                setPrevious(selectedVehicle);
            }
        }
    }, [previousSelection, select, selectedVehicle]);

    // Valuable code. Validating Datalist input.
    // When the value of the input changes...
    const inputValidation = (event) => {
        var optionFound = false,
            input = event.target,
            datalist = input.list;
        
        // Determine whether an option exists with the current value of the input.
        for (let j = 0; j < datalist.options.length; j++) {
            let optionValue = datalist.options[j].value;
            if(input.value.toUpperCase() === optionValue.toUpperCase()) {
                optionFound = true;
                setSelection(optionValue);
                break;
            }
        }
        // Use the setCustomValidity function of the validation API
        // to provide a user feedback if the value does not exist in the datalist.
        if(optionFound) {
            input.setCustomValidity('');
        } else {
            input.setCustomValidity('Please select a valid Vehicle');
        }
        input.reportValidity();
    }

    // Datalists must be given dynamic ID, even though in separate components. 
    return (
        <div key={distance} className="vehicle-list">
            <label htmlFor="vehicle-selector" className="sr-only">
            Choose a Vehicle from this list:
            </label>
            <input
            type="text"
            list={distance}
            name="vehicle-selector"
            placeholder="Pick a vehicle."
            onChange={inputValidation}
            autoCorrect="off"
            spellCheck="false"
            />
            <datalist id={distance}>{options}</datalist>
        </div>
    );
}

export default Presentational;