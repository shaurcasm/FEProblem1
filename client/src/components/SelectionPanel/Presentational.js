import React, { useEffect, useState } from 'react'
import PlanetList from '../PlanetList'
//import VehicleList from '../VehicleList'

// ToDo: Vehicles; Image Styling; Panel Styling.
const DEFAULT_SRC = "/images/misc/al-falcone.png";
const DEFAULT_ALT = "Target Queen Al Falcone";

const planetToImageSrcMatrix = {
    'Donlon': "/images/planets/donlon.png",
    'Enchai': "/images/planets/enchai.png",
    'Jebing': "/images/planets/jebing.png",
    'Lerbin': "/images/planets/lerbin.png",
    'Pingasor': "/images/planets/pingasor.png",
    'Sapir': "/images/planets/sapir.png"
}

// Might not need container if redux state not used.
const Presentational =  ({ planets, select }) => {
    // Any maintenance when component renders.
    useEffect(() => {

    });

    const [imageSource, setSource] = useState(DEFAULT_SRC);
    const [planetName, setName] = useState(DEFAULT_ALT);

    // Set image of the planet if available, else set default if no planet or no image.
    const changePlanetImage = (planet) => {
        var src = planetToImageSrcMatrix[planet];
        var imageNameRE = /(?<=\/)\w+(?=\.png$)/;
        var name = src.match(imageNameRE);

        if(imageNameRE.test(src)) {
            setSource(src);
            setName(name);
        }
        else {
            setSource(DEFAULT_SRC);
            setName(DEFAULT_ALT);
        }
    }
    // Add vehicle - ToDo.
    // Change image when a planet is selected. Function to change (this) Local state sent as prop to PlanetList.
    return (
        <div className="selection-panel">
            <img className="planet-image" src={imageSource} alt={planetName}></img>
            <div className="selectors-container">
                <PlanetList changeImage={changePlanetImage}/>
            </div>
        </div>
    )
}

export default Presentational;