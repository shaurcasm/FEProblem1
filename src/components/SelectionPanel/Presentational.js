import React, { useEffect, useState } from 'react';
import PlanetList from '../PlanetList';
import VehicleList from '../VehicleList';
import { DEFAULT_SRC, DEFAULT_ALT, DEFAULT_RANGE, planetToImageSrcMatrix } from '../../constants/misc.js';
import PropTypes from 'prop-types';
import './style.scss';

const Presentational =  ({ planets, vehicles, selectPlanets, selectVehicles, direction }) => {
    const [imageSource, setSource] = useState(DEFAULT_SRC);
    const [planetName, setName] = useState(DEFAULT_ALT);
    const [distanceToPlanet, setDistance] = useState(DEFAULT_RANGE);
    const [timeToPlanet, setTime] = useState(null);
    const reverseOrNot = direction || 'row';

    // Set image of the planet if available, else set default if no planet or no image.
    const changePlanetImage = planet => {
        var src = planetToImageSrcMatrix[planet];
        var imageNameRE = /(?<=\/)\w+(?=\.png$)/;

        if(imageNameRE.test(src)) {
            setSource(src);
            setName(planet);
        }
        else {
            setSource(DEFAULT_SRC);
            setName(DEFAULT_ALT);
        }
    }

    useEffect(() => {
        setDistance(() => {
            let selectedPlanet = planets.filter(planet => planet.name === planetName);
            return selectedPlanet.length === 1 ? selectedPlanet[0].distance : DEFAULT_RANGE;
        });
    }, [planetName, planets]);

    // Reverse flex-direction for alternate panels. Display time to reach planet when both planet, vehicle selected
    // and time-to-reach is calculated.
    return (
        <div style={{flexDirection: reverseOrNot}} className="selection-panel">
            <div className="image-container">
                <img className="planet-image" src={imageSource} alt={planetName}></img>
            </div>
            { timeToPlanet &&
                <div className="time-container">
                    Touchdown in ...
                    <p className="time">{timeToPlanet} units</p>
                </div>
            }
            <div className="selectors-container">
                <PlanetList planets={planets} select={selectPlanets} changeImage={changePlanetImage} />
                <VehicleList vehicles={vehicles} select={selectVehicles} distance={distanceToPlanet} changeTime={setTime} />
            </div>
        </div>
    )
}

Presentational.propTypes = {
    planets: PropTypes.array,
    vehicles: PropTypes.array,
    selectPlanets: PropTypes.objectOf(PropTypes.func),
    selectVehicles: PropTypes.objectOf(PropTypes.func),
    direction: PropTypes.string
};

export default Presentational;