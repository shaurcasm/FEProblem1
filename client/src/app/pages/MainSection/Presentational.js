//Presentational code for Main Section
import React, { useEffect, useState } from 'react';
import SelectionPanel from '../../../components/SelectionPanel';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSun } from '@fortawesome/free-solid-svg-icons';
import { Redirect } from 'react-router-dom';
import fetch from 'node-fetch';
import './style.scss';
import PropTypes from 'prop-types';

const Presentational = ({ loading, error, selectionPanelVisibility, selectedOptions, fetchPlanets, fetchVehicles }) => {
    const [redirect, setRedirect] = useState();
    const [result, setResult] = useState();

    // ComponentDidMount() but for functional component.
    useEffect(() => {
        fetchPlanets();
        fetchVehicles();
    }, [fetchPlanets, fetchVehicles]);

    const onSubmit = () => {
        // Process user input and send to route for result.
        const requestOptions = {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify(selectedOptions)
        };
        
        fetch('/api/getResult', requestOptions)
            .then(async response => {
                const data = await response.json();

                // Check for error response
                if(!response.ok) {
                    // Get error message from body or default to response status
                    const error = (data && data.message) || response.status;
                    return Promise.reject(error);
                }

                console.log("In React Result = " + JSON.stringify(data));
                setResult(data);        // set data state to be provided when redirecting to result page
                setRedirect('/result'); // set local state to initiate rerender and redirect to result with above data.
                return data;
            })
            .catch(error => {
                setResult({ error: error.toString() });
                setRedirect('/result');
                console.log('Error = ' + error)
            });  // Display error.
    }

    // Display any Errors while fetching
    if (error) {
        return (
            <div id='error'>
                <span id='error-code'>{error.status}</span>
                Error! {error.message}
            </div>
        )
    }
    // Luxury: Display something while fetching, if wanted.
    if (loading) {
        return ( 
            <div id='loading'>
                <FontAwesomeIcon icon={faSun} color='orange' size='6x' spin />
            </div>
        )
    }
    // Redirect to result page with received result data.
    if (redirect) {
        return <Redirect to={{
            pathname: redirect,
            state: { result: result }
        }} />
    }

    // Display SectionalPanels one by one after checking from selector; Reverse display direction every alternate panel.
    return (
        <section className="main">
            <div id='planet-one' className="selection-container" style={{ display: "flex" }} >
                <SelectionPanel key='1' />
            </div>
            <div id='planet-two' className="selection-container" style={{ display: selectionPanelVisibility.second }} >
                <SelectionPanel direction="row-reverse" key='2' />
            </div>
            <div id='planet-three' className="selection-container" style={{ display: selectionPanelVisibility.third }} >
                <SelectionPanel key='3' />
            </div>
            <div id='planet-four' className="selection-container" style={{ display: selectionPanelVisibility.fourth }} >
                <SelectionPanel direction="row-reverse" key='4' />
            </div>
            <div className='button-container'>
                <button id='submit-button' type='submit' disabled={!selectionPanelVisibility.submitButton} onClick={onSubmit}>Launch</button>
            </div>
        </section>
    );
}

// PropType description to ensure structure.
Presentational.propTypes = {
    loading: PropTypes.bool,
    error: PropTypes.object,
    selectionPanelVisibility: PropTypes.object.isRequired,
    selectedOptions: PropTypes.exact({
        'planet_names': PropTypes.array,
        'vehicle_names': PropTypes.array
    }),
    fetchPlanets: PropTypes.func,
    fetchVehicles: PropTypes.func
};

export default Presentational;