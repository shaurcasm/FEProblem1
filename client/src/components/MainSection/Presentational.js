//Presentational code for Main Section

import React, { useEffect, useRef } from 'react';
import SelectionPanel from '../SelectionPanel';
import './style.scss';
import fetch from 'node-fetch';
//import PropTypes from 'prop-types';

const Presentational = ({ loading, error, selectionPanelVisibility, selectedOptions, fetchPlanets, fetchVehicles }) => {
    const resultRef = useRef(null);

    // ComponentDidMount() but for functional component.
    useEffect(() => {
        //console.log("Fetching...");
        fetchPlanets();
        fetchVehicles();
        //console.log('Done');
    }, [fetchPlanets, fetchVehicles]);

    const onSubmit = () => {
        // Process user input and send to route for result.
        // const token = importedTokenAPI()
        // Handle 4xx and 5xx errors for fetch.
        const handleErrors = (response) => {
            if(!response.ok) {
                throw Error(response.statusText);
            }
            return response;
        };

        //console.log("Selected Options = " + JSON.stringify(selectedOptions));   // Convert object to JSON to display. 

        const requestOptions = {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify(selectedOptions)
        }

        fetch('/api/getResult', requestOptions)
            .then(handleErrors)
            .then(res => res.json())
            .then(data => {
                console.log("In React Result = " + JSON.stringify(data));
                if(data.error) {
                    resultRef.current.innerHTML = data.error;
                }
                else {
                    if((data.status === 'success') && data["planet_name"]) {
                        resultRef.current.innerHTML = `It's a success. Found Queen in <span style='color: red'>${data["planet_name"]}</span>.`;
                    }
                    else {
                        resultRef.current.innerHTML = `Mission failed. We'll get her next <span style='color: red'>time</span>.`;
                    }
                }
            })
            .catch(error => console.log('Error = ' + error))  // Display error.

        //console.log('Result in onSubmit = ' + JSON.stringify(result));
    }

    // Display any Errors while fetching
    if (error) {
        return <div id='error'>Error! {error.message}</div>
    }
    // Luxury: Display something while fetching, if wanted.
    if (loading) {
        return <div id='loading'>Loading...</div>
    }

    // Display each latter SectionalPanel after checking from selector.
    return (
        <section className="main">
            <h2>Choose 4 planets and vehicles: Find Queen Al Falcone.</h2>
            <div id='planet-one' className="selection-container" style={{ display: "flex" }} >
                <SelectionPanel key='1' />
            </div>
            <div id='planet-two' className="selection-container" style={{ display: selectionPanelVisibility.second }} >
                <SelectionPanel direction="row-reverse" key='2' />
            </div>
            <div id='planet-three' className="selection-container" style={{ display: selectionPanelVisibility.third }} >
                <SelectionPanel key='3' />
            </div>
            <div id='planet-four' className="selection-container" style={{ display: selectionPanelVisibility.fourth }} >
                <SelectionPanel direction="row-reverse" key='4' />
            </div>
            <button id='submit-button' type='submit' disabled={selectionPanelVisibility.submitButton} onClick={onSubmit}>Submit</button>
            <div ref={resultRef} className='result-container'></div>
        </section>
    );
}

export default Presentational;