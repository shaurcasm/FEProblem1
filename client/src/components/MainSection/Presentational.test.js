import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

// Component to be tested
import Presentational from './Presentational.js';

describe('Presentational <MainSection />', () => {

    const mockFetchPlanets = jest.fn();
    const mockFetchVehicles = jest.fn();
    const mockOptions = {
        'planet_names': [],
        'vehicle_names': []
    }

    describe('render()', () => {

        test('Renders one Panel at default', () => {

            var testVisibility = {
                second: 'none',
                third: 'none',
                fourth: 'none',
                submitButton: false
            };
            const component = shallow(<Presentational loading={false} error={null} selectionPanelVisibility={testVisibility} selectedOptions={mockOptions} fetchPlanets={mockFetchPlanets} fetchVehicle={mockFetchVehicles} />);
            
            expect(toJson(component)).toMatchSnapshot();
        });

        test('Renders two panels when one planets selected', () => {

            var testVisibility = {
                second: 'flex',
                third: 'none',
                fourth: 'none',
                submitButton: false
            };
            const component = shallow(<Presentational loading={false} error={null} selectionPanelVisibility={testVisibility} selectedOptions={mockOptions} fetchPlanets={mockFetchPlanets} fetchVehicle={mockFetchVehicles} />);

            expect(toJson(component)).toMatchSnapshot();
        });

        test('Renders three panels when two planets selected', () => {

            var testVisibility = {
                second: 'flex',
                third: 'flex',
                fourth: 'none',
                submitButton: false
            };
            const component = shallow(<Presentational loading={false} error={null} selectionPanelVisibility={testVisibility} selectedOptions={mockOptions} fetchPlanets={mockFetchPlanets} fetchVehicle={mockFetchVehicles} />);

            expect(toJson(component)).toMatchSnapshot();
        });

        test('Renders four panels when three planets selected', () => {

            var testVisibility = {
                second: 'flex',
                third: 'flex',
                fourth: 'flex',
                submitButton: false
            };
            const component = shallow(<Presentational loading={false} error={null} selectionPanelVisibility={testVisibility} selectedOptions={mockOptions} fetchPlanets={mockFetchPlanets} fetchVehicle={mockFetchVehicles} />);

            expect(toJson(component)).toMatchSnapshot();
        });

        test('Renders all panels and enables submit Button when four planets selected', () => {

            var testVisibility = {
                second: 'flex',
                third: 'flex',
                fourth: 'flex',
                submitButton: true
            };
            const component = shallow(<Presentational loading={false} error={null} selectionPanelVisibility={testVisibility} selectedOptions={mockOptions} fetchPlanets={mockFetchPlanets} fetchVehicle={mockFetchVehicles} />);

            expect(toJson(component)).toMatchSnapshot();
        });
    });
});