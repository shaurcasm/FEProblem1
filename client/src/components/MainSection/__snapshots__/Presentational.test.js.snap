// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Presentational <MainSection /> render() Renders all panels and enables submit Button when four planets selected 1`] = `
<section
  className="main"
>
  <h2>
    Choose 4 planets and vehicles: Find Queen Al Falcone.
  </h2>
  <div
    className="selection-container"
    id="planet-one"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <Connect(Presentational)
      key="1"
    />
  </div>
  <div
    className="selection-container"
    id="planet-two"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <Connect(Presentational)
      direction="row-reverse"
      key="2"
    />
  </div>
  <div
    className="selection-container"
    id="planet-three"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <Connect(Presentational)
      key="3"
    />
  </div>
  <div
    className="selection-container"
    id="planet-four"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <Connect(Presentational)
      direction="row-reverse"
      key="4"
    />
  </div>
  <button
    disabled={false}
    id="submit-button"
    onClick={[Function]}
    type="submit"
  >
    Submit
  </button>
  <div
    className="result-container"
  />
</section>
`;

exports[`Presentational <MainSection /> render() Renders four panels when three planets selected 1`] = `
<section
  className="main"
>
  <h2>
    Choose 4 planets and vehicles: Find Queen Al Falcone.
  </h2>
  <div
    className="selection-container"
    id="planet-one"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <Connect(Presentational)
      key="1"
    />
  </div>
  <div
    className="selection-container"
    id="planet-two"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <Connect(Presentational)
      direction="row-reverse"
      key="2"
    />
  </div>
  <div
    className="selection-container"
    id="planet-three"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <Connect(Presentational)
      key="3"
    />
  </div>
  <div
    className="selection-container"
    id="planet-four"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <Connect(Presentational)
      direction="row-reverse"
      key="4"
    />
  </div>
  <button
    disabled={true}
    id="submit-button"
    onClick={[Function]}
    type="submit"
  >
    Submit
  </button>
  <div
    className="result-container"
  />
</section>
`;

exports[`Presentational <MainSection /> render() Renders one Panel at default 1`] = `
<section
  className="main"
>
  <h2>
    Choose 4 planets and vehicles: Find Queen Al Falcone.
  </h2>
  <div
    className="selection-container"
    id="planet-one"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <Connect(Presentational)
      key="1"
    />
  </div>
  <div
    className="selection-container"
    id="planet-two"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <Connect(Presentational)
      direction="row-reverse"
      key="2"
    />
  </div>
  <div
    className="selection-container"
    id="planet-three"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <Connect(Presentational)
      key="3"
    />
  </div>
  <div
    className="selection-container"
    id="planet-four"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <Connect(Presentational)
      direction="row-reverse"
      key="4"
    />
  </div>
  <button
    disabled={true}
    id="submit-button"
    onClick={[Function]}
    type="submit"
  >
    Submit
  </button>
  <div
    className="result-container"
  />
</section>
`;

exports[`Presentational <MainSection /> render() Renders three panels when two planets selected 1`] = `
<section
  className="main"
>
  <h2>
    Choose 4 planets and vehicles: Find Queen Al Falcone.
  </h2>
  <div
    className="selection-container"
    id="planet-one"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <Connect(Presentational)
      key="1"
    />
  </div>
  <div
    className="selection-container"
    id="planet-two"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <Connect(Presentational)
      direction="row-reverse"
      key="2"
    />
  </div>
  <div
    className="selection-container"
    id="planet-three"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <Connect(Presentational)
      key="3"
    />
  </div>
  <div
    className="selection-container"
    id="planet-four"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <Connect(Presentational)
      direction="row-reverse"
      key="4"
    />
  </div>
  <button
    disabled={true}
    id="submit-button"
    onClick={[Function]}
    type="submit"
  >
    Submit
  </button>
  <div
    className="result-container"
  />
</section>
`;

exports[`Presentational <MainSection /> render() Renders two panels when one planets selected 1`] = `
<section
  className="main"
>
  <h2>
    Choose 4 planets and vehicles: Find Queen Al Falcone.
  </h2>
  <div
    className="selection-container"
    id="planet-one"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <Connect(Presentational)
      key="1"
    />
  </div>
  <div
    className="selection-container"
    id="planet-two"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <Connect(Presentational)
      direction="row-reverse"
      key="2"
    />
  </div>
  <div
    className="selection-container"
    id="planet-three"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <Connect(Presentational)
      key="3"
    />
  </div>
  <div
    className="selection-container"
    id="planet-four"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <Connect(Presentational)
      direction="row-reverse"
      key="4"
    />
  </div>
  <button
    disabled={true}
    id="submit-button"
    onClick={[Function]}
    type="submit"
  >
    Submit
  </button>
  <div
    className="result-container"
  />
</section>
`;
